btf/core.go

```diff
@@ -164,7 +164,7 @@ func (k coreKind) String() string {
 //
 // Fixups are returned in the order of relos, e.g. fixup[i] is the solution
 // for relos[i].
-func CORERelocate(relos []*CORERelocation, target *Spec, bo binary.ByteOrder) ([]COREFixup, error) {
+func CORERelocate(relos []*CORERelocation, target *Spec, bo binary.ByteOrder, typeID func(Type) (TypeID, error)) ([]COREFixup, error) {
```

Comment:
Would it be wrong to make CORERelocate take a Builder here? I understand this way the changes are kept minimal, but it doesn't convey clearly what (potentially) happens behind the scenes, being adding a type to the BTF blob. AFAICS, we only pass `spec.TypeID()` here in tests.